nautobot/core/views/generic.py

```diff
@@ -345,6 +345,12 @@ def post(self, request, *args, **kwargs):
             try:
                 with transaction.atomic():
                     object_created = not form.instance.present_in_database
+                    if hasattr(form.instance, "set_dryrun"):
+                        if "_dryrun_create" in request.POST:
+                            form.instance.set_dryrun(mode="create")
+                        elif "_dryrun_update" in request.POST:
+                            form.instance.set_dryrun(mode="update")
+
```

Comment:
Thats true, i've moved it to `forms.GitRepositoryForm.clean()`, I don't fully believe this might be the best place because it has noting to do with validation/cleaning,
but i also don't think overriding `GitRepositoryEditView.post()` would be the best approach